atmo/coordinator/sequence/sequence.go

```diff
@@ -221,13 +234,19 @@ func (seq *Sequence) HandleStepErrs(results []FnResult, step executable.Executab
 //
 // sequence.Execute -> exec.do -> handleMessage (n times) -> .do returns to .Execute
 func (seq *Sequence) handleMessage(msg grav.Message) error {
+	seq.lock.Lock()
+	defer seq.lock.Unlock()
+
 	result := FnResult{}
 	if err := json.Unmarshal(msg.Data(), &result); err != nil {
 		return errors.Wrap(err, "failed to Unmarshal FnResult")
 	}
 
+	seq.log.Info("handleMessage recieved", msg.UUID(), "(", msg.ParentID(), ")")
```

Comment:
Do we want to keep this message?

Reply:
Yes, but I downgraded it to debug level.